lib/cadet/assessments/assessments.ex

```diff
@@ -448,59 +449,40 @@ defmodule Cadet.Assessments do
           assessment: a
       })
 
-    case role do
-      @grading_role ->
-        students = Cadet.Accounts.Query.students_of(grader)
+    cond do
+      role in @grading_roles ->
+        {:ok, submissions_by_group(grader, submission_query)}
 
+      role in @see_all_submissions_roles ->
         submissions =
-          submission_query
-          |> join(:inner, [s], st in subquery(students), s.student_id == st.id)
-          |> Repo.all()
+          (group_only && role != @admin_role && submissions_by_group(grader, submission_query)) ||
```

Comment:
The intent of this line is not too clear. Perhaps this is better?
```elixir
if group_only and role != @admin_role
  submissions_by_group(grader, submission_query)
else
  Repo.all(submission_query)
```

Reply:
Not sure why you need the `role != @admin-role` though? An admin would never be part of a group so he would of course get 0 results. I think that's logical?

Reply:
It was because the front-end by default will query `/grading/?group=true`. So, an admin will not see any results until he ticks the checkbox. 

Is there a better way to do that?

Reply:
Or instead, you can make the change in frontend instead to request the right thing based on the role.

I'm pretty sure GET /user gives back role

By default show all submission should be checked for admin right?

Cos if not the behaviour would be inconsistent. It's logical that specifying group to be true for any non student not the leader of any group to have 0 submission.